src/updaters/pubspec-yaml.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Update, UpdateOptions, VersionsMap} from './update';
+import {GitHubFileContents} from '../github';
+import {logger} from '../util/logger';
+
+export class PubspecYaml implements Update {
+  path: string;
+  changelogEntry: string;
+  version: string;
+  versions?: VersionsMap;
+  packageName: string;
+  create: boolean;
+  contents?: GitHubFileContents;
+
+  constructor(options: UpdateOptions) {
+    this.create = false;
+    this.path = options.path;
+    this.changelogEntry = options.changelogEntry;
+    this.version = options.version;
+    this.packageName = options.packageName;
+    this.contents = options.contents;
+  }
+
+  updateContent(content: string): string {
+    const oldVersion = content.match(/version: ([0-9\.]+)\+([0-9]+)/);
+    let buildNumber = '1';
+
+    if (oldVersion) {
+      buildNumber = `${oldVersion[2]}`;
+      logger.info(
+        `updating ${this.path} from ${oldVersion[1]}+${buildNumber} to ${this.version}+${buildNumber}`
+      );
+    }
+    return content.replace(
+      /version: ([0-9\.]+)\+([0-9]+)/,
+      `version: ${this.version}+${buildNumber}`
+    );
+  }
```

Comment:
If I'm reading this right, say the current version is `1.2.3+45` and we release a new feature (`feat`), then the new version would be `1.3.0+45`. Is this intentional?

Reply:
@chingor13 Yeah, I intended to keep the build number and update only the version.